tests/libc/enc/CMakeLists.txt

```diff
@@ -62,51 +41,115 @@ target_compile_options(libc_enc PRIVATE
+add_library(libctest-support
+    enc.c
+    helpers.c
+    libc_t.c
+    ${test_extras})
+
+target_include_directories(libctest-support PRIVATE ${LIBC_INCLUDE_DIRECTORIES})
+target_link_libraries(libctest-support PRIVATE oecore)
+target_link_libraries(libctest-support PRIVATE oelibc)
+target_compile_options(libctest-support PRIVATE ${LIBC_COMPILE_OPTIONS})
+
+set (LIBC_ENC_REQUIRED_LIBRARIES
+    libctest-support
+    oelibc
+    oesyscall
+    oehostresolver
+    oehostfs
+    oehostsock)
+
+# A function for compiling a single libc test in its own enclave binary.
+# This is only used when libc tests are run individually by defining
+# SEPERATE_LIBC_TESTS=TRUE
+function(add_libc_test_enc NAME CFILE)
+    # Generate a new uuid per enclave for trustzone
+    string(UUID enclave_uuid
+           NAMESPACE "d7fe296a-24e9-46d1-aa78-9c7395082a41"
+           NAME libc-${NAME}_enc
+           TYPE SHA1)
+
+    add_enclave(TARGET libc-${NAME}_enc UUID ${enclave_uuid} SOURCES
+        ${PROJECT_SOURCE_DIR}/${CFILE}
+        run_tests_single.c
+        libc_t.c)
+
+    # add_enclave modifies CMAKE_C_LINK_EXECUTABLE, CMAKE_CXX_LINK_EXECUTABLE,
+    # and CMAKE_EXE_LINKER_FLAGS. Propogate them to the parent scope
+    set(CMAKE_C_LINK_EXECUTABLE ${CMAKE_C_LINK_EXECUTABLE} PARENT_SCOPE)
+    set(CMAKE_CXX_LINK_EXECUTABLE ${CMAKE_CXX_LINK_EXECUTABLE} PARENT_SCOPE)
+    set(CMAKE_EXE_LINKER_FLAGS ${CMAKE_EXE_LINKER_FLAGS} PARENT_SCOPE)
+
+    target_compile_options(libc-${NAME}_enc PRIVATE ${LIBC_COMPILE_OPTIONS} -D__TEST__="${NAME}")
+    target_include_directories(libc-${NAME}_enc PRIVATE ${LIBC_INCLUDE_DIRECTORIES})
+    target_link_libraries(libc-${NAME}_enc ${LIBC_ENC_REQUIRED_LIBRARIES})
+endfunction(add_libc_test_enc)
+
+if (OE_SEPARATE_LIBC_TESTS)
+    foreach (test_file ${LIBC_TESTS})
+        get_testcase_name(${test_file} name "../../")
+        add_libc_test_enc("${name}" "${test_file}")
+    endforeach ()
+else ()
+    foreach (test_file ${LIBC_TESTS})
+        string(MAKE_C_IDENTIFIER ${test_file} test_name)
+        configure_file(include.c.in ${CMAKE_CURRENT_BINARY_DIR}/${test_file})
+        list(APPEND test_dependencies ${CMAKE_CURRENT_BINARY_DIR}/${test_file})
+        string(APPEND libc_tests "\
+        extern int ${test_name}(int argc, const char* argv[]);\n\
+        ret += run_single_test(\"${test_name}\", ${test_name});\n")
+    endforeach ()
+
+    configure_file(run_tests_all.c.in run_tests_all.c)
+    add_enclave(TARGET libc_enc UUID d7fe296a-24e9-46d1-aa78-9c7395082a41 SOURCES
+        ${CMAKE_CURRENT_BINARY_DIR}/run_tests_all.c
+        ${test_dependencies}
+        libc_t.c)
+
+    target_compile_options(libc_enc PRIVATE ${LIBC_COMPILE_OPTIONS})
+    target_include_directories(libc_enc PRIVATE ${LIBC_INCLUDE_DIRECTORIES})
+    target_link_libraries(libc_enc ${LIBC_ENC_REQUIRED_LIBRARIES})
+endif ()
+
+# On windows, build the enclave support library with clangw
+if (WIN32)
+    maybe_build_using_clangw(libunwindtest-${NAME}_enc)
```

Comment:
I guess we can remove this if logic as `add_enclave` should handle the `clangw` configuration (also the `libunwindtest-${NAME}_enc` is incorrect).